cli/start.go

```diff
@@ -84,23 +83,32 @@ func MakeStartCommand() *cobra.Command {
 			}
 
 			if cfg.GetString("datastore.store") != configStoreMemory {
-				// It would be ideal to not have the key path tied to the datastore.
-				// Running with memory store mode will always generate a random key.
-				// Adding support for an ephemeral mode and moving the key to the
-				// config would solve both of these issues.
 				rootDir := mustGetContextRootDir(cmd)
-				key, err := loadOrGeneratePrivateKey(filepath.Join(rootDir, "data", "key"))
-				if err != nil {
-					return err
-				}
-				netOpts = append(netOpts, net.WithPrivateKey(key))
-
 				// TODO-ACP: Infuture when we add support for the --no-acp flag when admin signatures are in,
 				// we can allow starting of db without acp. Currently that can only be done programmatically.
 				// https://github.com/sourcenetwork/defradb/issues/2271
 				dbOpts = append(dbOpts, db.WithACP(rootDir))
 			}
 
+			if !cfg.GetBool("keyring.disabled") {
+				keyring, err := openKeyring(cmd, cfg.GetString("keyring.path"))
+				if err != nil {
+					return err
+				}
+
+				peerKey, err := loadOrGenerateEd25519(keyring, peerKeyName)
+				if err != nil {
+					return err
+				}
+				netOpts = append(netOpts, net.WithPrivateKey(peerKey))
+
+				encryptionKey, err := loadOrGenerateAES256(keyring, badgerEncryptionKeyName)
+				if err != nil {
+					return err
+				}
+				storeOpts = append(storeOpts, node.WithEncryptionKey(encryptionKey))
```

Comment:
question: This implies that if the keyring is enabled, then we also use use at-rest encryption. Should these technically be separate config options?

Reply:
Good catch! I've added a `datastore.encryptionDisabled` config item.

Reply:
Update: I've made the key optional instead of the config option.